dango/types/src/lending.rs

```diff
@@ -10,27 +13,56 @@ pub static NAMESPACE: LazyLock<Part> = LazyLock::new(|| Part::new_unchecked("len
 /// Sub-namespace that liquidity share tokens will be minted under.
 pub static SUBNAMESPACE: LazyLock<Part> = LazyLock::new(|| Part::new_unchecked("pool"));
 
-/// Configurations and state of a market.
-#[grug::derive(Serde, Borsh)]
-pub struct Market {
-    // TODO
+// -------------------------------- Bounds -------------------------------------
+
+/// Defines the bounds for a loan-to-value ratio: 0 < LoanToValue < 1.
+#[grug::derive(Serde)]
+pub struct LoanToValueBounds;
+impl Bounds<Udec128> for LoanToValueBounds {
+    const MAX: Option<Bound<Udec128>> = Some(Bound::Exclusive(Udec128::ONE));
+    const MIN: Option<Bound<Udec128>> = Some(Bound::Exclusive(Udec128::ZERO));
 }
+/// A decimal bounded by the loan-to-value bounds.
+pub type LoanToValue = Bounded<Udec128, LoanToValueBounds>;
 
-/// A set of updates to be applied to a market.
+/// Defines the bounds for a collateral power: 0 < CollateralPower <= 1.
 #[grug::derive(Serde)]
-pub struct MarketUpdates {
+pub struct CollateralPowerBounds;
+impl Bounds<Udec128> for CollateralPowerBounds {
+    const MAX: Option<Bound<Udec128>> = Some(Bound::Inclusive(Udec128::ONE));
+    const MIN: Option<Bound<Udec128>> = Some(Bound::Exclusive(Udec128::ZERO));
+}
+/// A decimal bounded by the collateral power bounds.
+pub type CollateralPower = Bounded<Udec128, CollateralPowerBounds>;
+
+// -------------------------------- Market -------------------------------------
+
+/// Configurations and state of a market (borrowable assets).
+#[optional_struct(MarketUpdates)]
+#[derive(
+    Default, Clone, Debug, PartialEq, Eq, Serialize, Deserialize, BorshSerialize, BorshDeserialize,
+)]
+pub struct Market {
     // TODO
 }
 
+// -------------------------------- Messages -----------------------------------
 #[grug::derive(Serde)]
 pub struct InstantiateMsg {
-    pub markets: BTreeMap<Denom, MarketUpdates>,
+    pub markets: BTreeMap<Denom, Market>,
 }
 
 #[grug::derive(Serde)]
 pub enum ExecuteMsg {
     /// Apply updates to markets.
     UpdateMarkets(BTreeMap<Denom, MarketUpdates>),
+    /// Set the collateral power for a denom.
+    SetCollateralPower {
+        denom: Denom,
+        power: CollateralPower,
+    },
+    /// Delist a collateral token. Removes it from the collateral power map.
+    DelistCollateral { denom: Denom },
```

Comment:
i think we should put collateral powers in the app config. in the case we don't need to implement these two methods

Reply:
Sure. Can move it

Reply:
What about Markets? What are the pros and cons with app config?